### PR TITLE
T6315: added codeql reusable workflow

### PR DESCRIPTION
**Pull request by kumvijaya**
## Change Summary

Added codeql reusable workflow

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Other (please describe):

## Related PR(s)

https://vyos.dev/T6315

## Component(s) name

Workflow
## Proposed changes

Added reusable github action workflow for codeql analysis.

## Smoketest result

Call this workflow from any repo's workflow. Example
```
name: "Perform CodeQL Analysis"

on:
  workflow_dispatch:
  push:
    branches: [ "current", "sagitta", "equuleus" ]
  pull_request:
    # The branches below must be a subset of the branches above
    branches: [ "current" ]
  schedule:
    - cron: '22 10 * * 0'

permissions:
  actions: read
  contents: read
  security-events: write

jobs:
  codeql-analysis-call:
    uses: vyos/vyos-github-actions/.github/workflows/codeql-analysis.yml@main
    secrets: inherit
    with:
      languages: "['python']"

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**]
- [x] I have run the components [**SMOKETESTS**]
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+<!---
+What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
+NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
+the box, please use [x]
+-->
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Code style update (formatting, renaming)
+- [ ] Refactoring (no functional changes)
+- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
```
Comment (sever-sever) [inline .github/PULL_REQUEST_TEMPLATE.md]: There won't be Migrations from the old Vyatta

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+<!---
+Please describe in detail how you tested your changes. Include details of your testing
+environment, and the tests you ran. When pasting configs, logs, shell output, backtraces,
+and other large chunks of text, surround this text with triple backtics
+```
+like this
+```
+-->
+
+## Smoketest result
```
Comment (sever-sever) [inline .github/PULL_REQUEST_TEMPLATE.md]: There won't be smoketests

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+```
+-->
+
+## Checklist:
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
+<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
+- [ ] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
+- [ ] I have linked this PR to one or more Phabricator Task(s)
+- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
```
Comment (sever-sever) [inline .github/PULL_REQUEST_TEMPLATE.md]: There won't be smoketests

File: .github/labeler.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+equuleus:
+  - any:
+    - base-branch: 'equuleus'
+current:
+  - any:
+    - base-branch: 'current'
+crux:
```
Comment (sever-sever) [inline .github/labeler.yml]: `crux` shouldn't be used, EOL

File: .github/workflows/pull-request-labels.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---
+name: Add pull request labels
+
+on:
+  pull_request_target:
+    branches:
+      - current
+      - crux
```
Comment (sever-sever) [inline .github/workflows/pull-request-labels.yml]: `crux` shouldn't be used, EOL

File: .github/workflows/pull-request-management.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+---
+name: Build Pull Request Package
+
+on:
+  pull_request:
+    branches:
+      - current
+      - crux
```
Comment (sever-sever) [inline .github/workflows/pull-request-management.yml]: `crux` shouldn't be used, EOL 
Need to add the `sagitta` branch

File: .github/workflows/pull-request-message-check.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
+---
+name: Check pull request message format
+
+on:
+  pull_request:
+    branches:
+      - current
+      - crux
```
Comment (sever-sever) [inline .github/workflows/pull-request-message-check.yml]: `crux` shouldn't be used, EOL
Need to add the `sagitta` branch

**Review by sever-sever — CHANGES_REQUESTED** [review]

**Comment by kumvijaya** [comment]
@sever-sever  Fixed review comments, Please check. Thanks

File: .github/workflows/pull-request-management.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  pull_request:
+    branches:
+      - current
+      - sagitta
+      - equuleus
+
+jobs:
+  j2lint:
+    name: Validate j2 files
+    runs-on: ubuntu-20.04
```
Comment (sever-sever) [inline .github/workflows/pull-request-management.yml]: Can we use `ubuntu-latest`?

File: .github/workflows/pull-request-message-check.yml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  pull_request:
+    branches:
+      - current
+      - sagitta
+      - equuleus
+
+jobs:
+  check-pr-title:
+    name: Check pull request title
+    runs-on: ubuntu-20.04
```
Comment (sever-sever) [inline .github/workflows/pull-request-message-check.yml]: Can we use `ubuntu-latest`?

**Comment by andamasov** [comment]
maybe it should be mentioned that all actions can use ubuntu-latest
except for actions that will involve deb package build, this must use Debian 
@sever-sever can you give indications of what branch corresponds to which Debian

**Review by sever-sever — APPROVED** [review]

**Comment by kumvijaya** [comment]
@andamasov Please approve and merge

**Review by andamasov — APPROVED** [review]
looks good to me

**Pull request merged**